### PR TITLE
Add release note for v1.22.4

### PR DESCRIPTION
**Pull request by lunny**
Wait ~#32507~ and backport of ~#32243~, ~#32186~, ~#32514~

### CODE REVIEW

**Comment by wxiaoguang** [comment]
> Wait #32507 and backport of #32415, #32325, #32243, #32186, #32514

Some of them are too large to backport, I do not think it should backport everything, there have been enough regressions.

**Review by silverwind — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,36 @@ This changelog goes through the changes that have been made in each release
 without substantial changes to our git log; to see the highlights of what has
 been added to each release, please refer to the [blog](https://blog.gitea.com).
 
+## [1.22.4](https://github.com/go-gitea/gitea/releases/tag/1.22.4) - 2024-11-14
+
+* SECURITY
+  * Refactor internal routers (partial backport, auth token const time comparing) (#32473) (#32479)
+* BUGFIXES
+  * Fix `missing signature key` error when pulling Docker images with `SERVE_DIRECT` enabled (#32365) (#32397)
```
Comment (kdumontnu) [inline CHANGELOG.md:15]:
```suggestion
  * Fix get reviewers fails when selecting user without pull request permissions unit (#32415) (#32616)
  * Fix adding index files to tmp directory (#32360) (#32593)
  * Fix PR creation on forked repositories via API (#31863) (#32591)
  * Fix missing menu tabs in organization project view page (#32313) (#32592)
  * Support HTTP POST requests to `/userinfo`, aligning to OpenID Core specification (#32578) (#32594)
  * Fix debian package clean up cron job (#32351) (#32590)
  * Fix GetInactiveUsers (#32540) (#32588)
  * Allow the actions user to login via the jwt token (#32527) (#32580)
  * Fix submodule parsing (#32571) (#32577)
  * Refactor find forks and fix possible bugs that weaken permissions check (#32528) (#32547)
  * Fix some places that don't respect org full name setting (#32243) (#32550)
  * Refactor push mirror find and add check for updating push mirror (#32539) (#32549)
  * Fix basic auth with webauthn (#32531) (#32536)
  * Fix artifact v4 upload above 8MB (#31664) (#32523)
  * Fix oauth2 error handle not return immediately (#32514) (#32516)
  * Fix action not triggered when commit message is too long (#32498) (#32507)
  * Fix `GetRepoLink` nil pointer dereference on dashboard feed page when repo is deleted with actions enabled (#32501) (#32502)
  * Fix `missing signature key` error when pulling Docker images with `SERVE_DIRECT` enabled (#32397) (#32397)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,36 @@ This changelog goes through the changes that have been made in each release
...
+  * Fix disable 2fa bug (#32320) (#32330)
+  * Always update expiration time when creating an artifact (#32281) (#32285)
+  * Fix null errors on conversation holder (#32258) (#32266) (#32282)
+  * Only rename a user when they should receive a different name (#32247) (#32249)
+  * Fix checkbox bug on private/archive filter (#32236) (#32240)
+  * Add a doctor check to disable the "Actions" unit for mirrors (#32424) (#32497)
+  * Quick fix milestone deadline 9999 (#32423)
+  * Make `show stats` work when only one file changed (#32244) (#32268)
+  * Make `owner/repo/pulls` handlers use "PR reader" permission (#32254) (#32265)
+  * Update scheduled tasks even if changes are pushed by "ActionsUser" (#32246) (#32252)
```
Comment (kdumontnu) [inline CHANGELOG.md:55]:
```suggestion
  * Update scheduled tasks even if changes are pushed by "ActionsUser" (#32246) (#32252)
* MISC
  * Remove unnecessary code: `GetPushMirrorsByRepoID` called on all repo pages (#32560) (#32567)
  * Improve some sanitizer rules (#32534)
  * Update nix development environment vor v1.22.x (#32495)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,36 @@ This changelog goes through the changes that have been made in each release
 without substantial changes to our git log; to see the highlights of what has
 been added to each release, please refer to the [blog](https://blog.gitea.com).
 
+## [1.22.4](https://github.com/go-gitea/gitea/releases/tag/1.22.4) - 2024-11-14
+
+* SECURITY
+  * Refactor internal routers (partial backport, auth token const time comparing) (#32473) (#32479)
```
Comment (kdumontnu) [inline CHANGELOG.md:11]:
```suggestion
  * Refactor internal routers (partial backport, auth token const time comparing) (#32473) (#32479)
* PERFORMANCE
  * Remove transaction for archive download (#32186) (#32520)
```

**Review by kdumontnu — COMMENTED** [review]
Update notes as of `cf1a38b03d`

**Review by appleboy — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Pull request merged**